hpolib/benchmarks/ml/svm_benchmark.py

```diff
@@ -0,0 +1,316 @@
+        and validation data set and evaluates the model on the X_test data set.
+
+        Parameters
+        ----------
+        configuration : Dict, CS.Configuration
+            Configuration for the SVM Model
+        fidelity: Dict, None
+            Fidelity parameters, check get_fidelity_space(). Uses default (max) value if None.
+        rng : np.random.RandomState, int, None,
+            Random seed for benchmark. By default the class level random seed.
+            To prevent overfitting on a single seed, it is possible to pass a
+            parameter ``rng`` as 'int' or 'np.random.RandomState' to this function.
+            If this parameter is not given, the default random state is used.
+        kwargs
+
+        Returns
+        -------
+        Dict -
+            function_value : X_test loss
+            cost : time to X_train and evaluate the model
+            info : Dict
+                train_valid_loss: Loss on the train+valid data set
+                fidelity : used fidelities in this evaluation
+        """
+        assert np.isclose(fidelity['dataset_fraction'], 1), \
+            f'Data set fraction must be 1 but was {fidelity["dataset_fraction"]}'
+
+        start_time = time.time()
+
+        self.rng = rng_helper.get_rng(rng=rng, self_rng=self.rng)
+
+        # Concatenate training and validation dataset
+        if isinstance(self.X_train, sparse.csr.csr_matrix) or isinstance(self.X_valid, sparse.csr.csr_matrix):
+            data = sparse.vstack((self.X_train, self.X_valid))
+        else:
+            data = np.concatenate((self.X_train, self.X_valid))
+        targets = np.concatenate((self.y_train, self.y_valid))
+
+        # Transform hyperparameters to linear scale
+        hp_c = np.exp(float(configuration['C']))
+        hp_gamma = np.exp(float(configuration['gamma']))
+
+        model = self.get_pipeline(hp_c, hp_gamma)
+        model.fit(data, targets)
+
+        # Compute validation error
+        train_valid_loss = 1 - self.accuracy_scorer(model, data, targets)
+
+        # Compute test error
+        test_loss = 1 - self.accuracy_scorer(model, self.X_test, self.y_test)
+
+        cost = time.time() - start_time
+
+        return {'function_value': test_loss,
+                "cost": cost,
+                'info': {'train_valid_loss': train_valid_loss,
+                         'fidelity': fidelity['dataset_fraction']}}
+
+    def get_pipeline(self, C: float, gamma: float) -> pipeline.Pipeline:
+        """ Create the scikit-learn (training-)pipeline """
+
+        model = pipeline.Pipeline([
+            ('preprocess_one_hot',
+             ColumnTransformer([
+                 ("categorical", OneHotEncoder(categories=self.categories, sparse=False), self.categorical_data),
+                 ("continuous", "passthrough", ~self.categorical_data)])),
+            ('svm',
+             svm.SVC(gamma=gamma, C=C, random_state=self.rng))
```

Comment:
Could you please add the cache size here as well and make this a class attribute set to the default value so one could easily change this from the outside.